Cmake 3.10 Boost error

### Bug Report
Building SDL Core with Ubuntu 18 default cmake version 3.10 causes build errors due to an issue with find boost module.

##### Reproduction Steps
1. Build sdl core

##### Expected Behavior
Core builds

##### Observed Behavior
Core build errors out with several messages similar to:

```
undefined reference to 'boost::detail::get_current_thread_data()'
```

##### OS & Version Information
* OS/Version: Ubuntu 18
* SDL Core Version: Develop

--- a/src/3rd_party/CMakeLists.txt
+++ b/src/3rd_party/CMakeLists.txt
@@ -221,6 +221,7 @@ else()
 endif()
 
 set(BOOST_ROOT ${3RD_PARTY_INSTALL_PREFIX})
+set(Boost_NO_BOOST_CMAKE ON)
 find_package(Boost 1.72.0 COMPONENTS system thread date_time filesystem regex) 
 if (NOT ${Boost_FOUND})
   set(BOOST_LIB_SOURCE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/boost_src)

